src/app/containers/PageViewAnalytics/labelHelpers/article.test.js

```diff
@@ -0,0 +1,189 @@
+const {
+  getPageIdentifier,
+  getOptimoUrn,
+  getLanguage,
+  getPromoHeadline,
+  getPublishedTime,
+  getThingAttributes,
+} = require('./article');
+
+describe('getPageIdentifier', () => {
+  const goodData = {
+    metadata: {
+      locators: {
+        optimoUrn: 'prefix:desiredValue',
+      },
+    },
+  };
+
+  const badData = {
+    metadata: {
+      locators: {
+        unknownUrn: 'prefix:missedValue',
+      },
+    },
+  };
+
+  it('should construct page identifier', () => {
+    const optimoUrn = getPageIdentifier('service', goodData);
+
+    expect(optimoUrn).toEqual('health::service.articles.desiredValue.page');
+  });
+
+  it('should use "unknown" if optimo id is unknown', () => {
+    const optimoUrn = getPageIdentifier('service', badData);
+
+    expect(optimoUrn).toEqual('health::service.articles.unknown.page');
+  });
+
+  it('should use null if service is null', () => {
+    const optimoUrn = getPageIdentifier(null, goodData);
+
+    expect(optimoUrn).toEqual('health::news.articles.desiredValue.page');
+  });
+});
+
+describe('getOptimoUrn', () => {
```

Comment:
this is an option to avoid duplication here

```
const metadataValueChecker = (
  metadataToCheck,
  method,
  goodData,
  goodReturnValue,
  badData,
  badReturnValue,
) => {
  describe(`get${metadataToCheck}`, () => {
    it('should find value in good data', () => {
      const result = method(goodData);

      expect(result).toEqual(goodReturnValue);
    });

    it('should return null in bad data', () => {
      const result = method(badData);

      expect(result).toEqual(badReturnValue);
    });
  });
};

metadataValueChecker(
  'OptimoUrn',
  getOptimoUrn,
  {
    metadata: {
      locators: {
        optimoUrn: 'desired value',
      },
    },
  },
  'desired value',
  {
    metadata: {
      locators: {
        unknownUrn: 'missed value',
      },
    },
  },
  null,
);

metadataValueChecker(
  'Language',
  getLanguage,
  {
    metadata: {
      passport: {
        language: 'desired value',
      },
    },
  },
  'desired value',
  {
    metadata: {
      locators: {
        unknownUrn: 'missed value',
      },
    },
  },
  null,
);

metadataValueChecker(
  'PromoHeadline',
  getPromoHeadline,
  {
    promo: {
      headlines: {
        seoHeadline: 'desired value',
      },
    },
  },
  'desired value',
  {
    promo: {
      headlines: {
        unknown: 'missed value',
      },
    },
  },
  null,
);
```

Reply:
Curious of others thoughts @sareh @pjlee11, ^ does reduce ~duplication~ repetition but maybe in exchange for readability? I want to avoid another set of the metadata tests which are pretty hard to follow.

Reply:
Agreed, I prefer the tests as-is. If it's length we're worried about, I'd sooner delete all the newlines (they just seem so unnecessary in functions this small!)